modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/service/test/NotificationQueueEntryLocalServiceTest.java

```diff
@@ -45,20 +56,119 @@ public class NotificationQueueEntryLocalServiceTest {
 	public static final AggregateTestRule aggregateTestRule =
 		new LiferayIntegrationTestRule();
 
+	@BeforeClass
+	public static void setUpClass() throws Exception {
+		_notificationRecipientSettings = Arrays.asList(
+			NotificationTemplateUtil.createNotificationRecipientSetting(
+				"bcc", "bcc@liferay.com"),
+			NotificationTemplateUtil.createNotificationRecipientSetting(
+				"cc", "cc@liferay.com"),
+			NotificationTemplateUtil.createNotificationRecipientSetting(
+				"from", "from@liferay.com"),
+			NotificationTemplateUtil.createNotificationRecipientSetting(
+				"fromName", "From Name"),
+			NotificationTemplateUtil.createNotificationRecipientSetting(
+				"to", "to@liferay.com"));
+	}
+	
 	@Test
 	public void testAddNotificationQueueEntry() throws Exception {
 		Assert.assertEquals(
 			0,
 			_notificationQueueEntryLocalService.
 				getNotificationQueueEntriesCount());
 
+		try {
+			_notificationQueueEntryLocalService.addNotificationQueueEntry(
+				NotificationTemplateUtil.createNotificationContext(
+					Arrays.asList(
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"fromName", "From name"),
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"to", "to@liferay.com")),
+					NotificationConstants.TYPE_EMAIL));
+
+			Assert.fail();
+		}
+		catch (NotificationRecipientSettingValueException.FromMustNotBeNull
+					notificationRecipientSettingValueException) {
+
+			Assert.assertEquals(
+				"From is null",
+				notificationRecipientSettingValueException.getMessage());
+		}
+
+		try {
+			_notificationQueueEntryLocalService.addNotificationQueueEntry(
+				NotificationTemplateUtil.createNotificationContext(
+					Arrays.asList(
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"from", "from@liferay.com"),
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"to", "to@liferay.com")),
+					NotificationConstants.TYPE_EMAIL));
+
+			Assert.fail();
+		}
+		catch (NotificationRecipientSettingValueException.FromNameMustNotBeNull
+					notificationRecipientSettingValueException) {
+
+			Assert.assertEquals(
+				"From name is null",
+				notificationRecipientSettingValueException.getMessage());
+		}
+
 		User user = TestPropsValues.getUser();
+
+		try {
+			_notificationQueueEntryLocalService.addNotificationQueueEntry(
+				NotificationTemplateUtil.createNotificationContext(
+					user, null, null, null, NotificationConstants.TYPE_EMAIL));
+
+			Assert.fail();
+		}
+		catch (NotificationQueueEntrySubjectException
+					notificationQueueEntrySubjectException) {
+
+			Assert.assertEquals(
+				"Subject is null",
+				notificationQueueEntrySubjectException.getMessage());
+		}
+
+		try {
```

Comment:
Hi @murilostodolni,
Since you added a lot of `assert.fail()`, I wonder if we could refactor this code to something similar that exists in 
ObjectEntryLocalServiceTest.java. In this class, there is the method [_assertFailure()](https://github.com/liferay/liferay-portal/blob/master/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java#L2134-L2147). Could you see its usage and check if we could do the same?

What do you think? :smile:

Reply:
Hi @marcelabc 

I agree! I will work on it.